src/cms/views/utils/search_content_ajax.py

```diff
@@ -91,35 +62,141 @@ def search_content_ajax(request, region_slug, language_slug):
 
-    if user.has_perm("cms.view_page"):
+    if user.has_perm("cms.view_page") and "page" in object_types:
+        object_types.remove("page")
         results.extend(
-            find_objects(
-                PageTranslation, "page", region, language_slug, query, hierarchical=True
+            format_object_translation(
+                obj,
+                "page",
+                cms_url=reverse(
+                    "edit_page",
+                    kwargs={
+                        "page_id": obj.page.id,
+                        "region_slug": region.slug,
+                        "language_slug": language_slug,
+                    },
+                ),
             )
+            for obj in PageTranslation.search(region, language_slug, query)
         )
 
-    if user.has_perm("cms.view_poi"):
+    if user.has_perm("cms.view_poi") and "poi" in object_types:
+        object_types.remove("poi")
         results.extend(
-            find_objects(POITranslation, "poi", region, language_slug, query)
+            format_object_translation(
+                obj,
+                "poi",
+                cms_url=reverse(
+                    "edit_poi",
+                    kwargs={
+                        "poi_id": obj.poi.id,
+                        "region_slug": region.slug,
+                        "language_slug": language_slug,
+                    },
+                ),
+            )
+            for obj in POITranslation.search(region, language_slug, query)
         )
 
+    if (
+        user.has_perm("cms.view_pushnotification")
+        and "push_notification" in object_types
+    ):
+        object_types.remove("push_notification")
+        results.extend(
+            {
+                "title": push_notification.title,
+                "url": None,
+                "type": "push_notification",
+                "cms_url": reverse(
+                    "edit_push_notification",
+                    kwargs={
+                        "push_notification_id": push_notification.id,
+                        "language_slug": language_slug,
+                        "region_slug": region_slug,
+                    },
+                ),
+            }
+            for push_notification in PushNotificationTranslation.search(
+                region, language_slug, query
+            )
+        )
+
+    if user.has_perm("cms.view_region") and "region" in object_types:
+        object_types.remove("region")
+        results.extend(
+            {
+                "title": region.name,
+                "url": None,
+                "type": "region",
+                "cms_url": reverse("edit_region", kwargs={"region_slug": region.slug}),
+            }
+            for region in Region.search(query)
+        )
+
+    if user.has_perm("auth.view_user") and "user" in object_types:
+        if (
+            user.is_superuser
+            or user.is_staff
+            or (region and region in user.profile.regions.all())
+        ):
```

Comment:
I'm not sure, this should already be checked by the decorator `@region_permission_required`, or am I missing something?

Reply:
No, I missed that...